Add ability to run via containers

hoang-himself opened the pull request:
Related #384, closes #386

Disclaimers:
- I have no experience with PHP
- I use `podman` and `podman-compose` instead of `docker` and `docker-compose`

---

With this PR, running `docker compose up -d` is enough. Or in the case of `podman`, `podman-compose up -d`.

The config file shows `-56,+56` due to change of line endings from CRLF to LF

---

Checklist:
- [x] Test with `podman-compose`
- [x] Test with `docker compose`
- [ ] Test for compatibility with other web servers like XAMPP, WAMP, etc. (Help needed)
- [ ] Update `README`s (Help needed)

digininja commented:
I know nothing about Docker so can't help with testing that bit
unfortunately. Can you find someone who can help you?

On Wed, 29 Mar 2023, 18:32 Hoàng, ***@***.***> wrote:

> Disclaimers:
>
>    - I have no experience with PHP
>    - I use podman and podman-compose instead of docker and docker-compose
>
> With this PR, running docker compose up -d is enough. Or in case of podman,
> podman-compose -f compose.yml up -d.
>
> Also, the config file shows -56,+56 due to change of line endings from
> CRLF to LF.
>
> Checklist:
>
>    - Containerfile
>    - compose.yml
>    - Test with podman-compose
>    - Test with docker compose (Help needed)
>    - Test for compatibility with other web servers like XAMPP, WAMP, etc.
>    (Help needed)
>    - Update READMEs (Help needed)
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/digininja/DVWA/pull/548
> Commit Summary
>
>    - 5bfd2ad
>    <https://github.com/digininja/DVWA/pull/548/commits/5bfd2ade20eab1c5507ba310bb51bb2d865652e7>
>    refactor: change quotes for literal strings
>    - fbb3b94
>    <https://github.com/digininja/DVWA/pull/548/commits/fbb3b94e9f4e083bbbb4d66d4c411c281a349e5c>
>    feat: allow reading variables from the environment
>    - f28dc23
>    <https://github.com/digininja/DVWA/pull/548/commits/f28dc237baeb379e51ab44f5515b4dd929603db7>
>    feat: add container files
>
> File Changes
>
> (4 files <https://github.com/digininja/DVWA/pull/548/files>)
>
>    - *A* .containerignore
>    <https://github.com/digininja/DVWA/pull/548/files#diff-2a6c4de4fd4d540549889f7844cbbdaf4babb4ae28c753020a21600201447210>
>    (7)
>    - *A* Containerfile
>    <https://github.com/digininja/DVWA/pull/548/files#diff-5fcdf9b4580789697d834d1456a22bcfaa236d668fc180cad4775afc36ed5914>
>    (5)
>    - *A* compose.yml
>    <https://github.com/digininja/DVWA/pull/548/files#diff-3493e6b5ddf34891e572f911db893efd9e46af828e011ea778a7c1eb64763588>
>    (34)
>    - *M* config/config.inc.php.dist
>    <https://github.com/digininja/DVWA/pull/548/files#diff-0c3c9e45825c5d2815e9dd22c7ed705c480f29117c1d64abb75b43576bcf1f59>
>    (112)
>
> Patch Links:
>
>    - https://github.com/digininja/DVWA/pull/548.patch
>    - https://github.com/digininja/DVWA/pull/548.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/digininja/DVWA/pull/548>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4SWJ44P2HNCJQFZQB733W6RW23ANCNFSM6AAAAAAWMF7HAM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

hoang-himself commented:
Yes, I know some people with `docker` readily available.
What about the last 2 items?

digininja commented:
If you are running it in docker then you don't need xampp or any of the
others do you?

And for the README, you'll need to write up how it works, I don't know
enough to write it. If you are worried about the quality of English, I'm
happy to edit it.

And feel free to contact the others.

On Wed, 29 Mar 2023, 18:44 Hoàng, ***@***.***> wrote:

> Yes, I know some people with docker readily available.
> What about the last 2 items?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/digininja/DVWA/pull/548#issuecomment-1489037726>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4SWLSF6QZWGL3WWB2Q4TW6RYHVANCNFSM6AAAAAAWMF7HAM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

hoang-himself commented:
Yes, with Docker available, we can drop all other installation methods.

digininja commented:
Not drop the other method, but you don't have to support them.

On Thu, 30 Mar 2023, 01:25 Hoàng, ***@***.***> wrote:

> Yes, with Docker available, we can drop all other installation methods.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/digininja/DVWA/pull/548#issuecomment-1489516594>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4SWKIIYQTLVUNZ3URLB3W6THIHANCNFSM6AAAAAAWMF7HAM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

hoang-himself commented:
I have removed the `vulnerables/web-dvwa` image and added instructions to run DVWA via Docker in the README. I don't understand other languages so I can't update their READMEs.

digininja commented:
The big thing you have to think about when making any changes is that a lot of the users have no idea how to use Linux or how to run things on the command line and so you have to keep it as simple as possible.

I ask users for access and error logs when they report bugs, how do they get them out of the container so they can send them over? That will need documenting.

It is looking good so far, but needs a bit more work.

hoang-himself commented:
I think that's because I have wei/pull app enabled. I'll fix the PR soon.

digininja commented:
No rush. Just wanted to check you hadn't accidentally messed up Git and
lost any changes. I do it all the time

On Mon, 3 Apr 2023 at 09:17, Hoàng ***@***.***> wrote:

> I think that's because I have wei/pull app enabled. I'll fix the PR soon.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/digininja/DVWA/pull/548#issuecomment-1493883697>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4SWIJUT7JX4OE6D56LYTW7KBTPANCNFSM6AAAAAAWMF7HAM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

hoang-himself commented:
I see that you have a section for logs in the bug report template. Do you want the template updated or include a new section in README?
